code/modules/mob/living/carbon/human/emote.dm

```diff
@@ -443,6 +445,17 @@
 								message = "<B>[src]</B> does a flip!"
 								SpinAnimation(5,1)
 
+		if("spin", "spins")
+			if(!restrained() && !lying)
+				if(prob(5))
+					INVOKE_ASYNC(src, .proc/spin, 30, 1)
```

Comment:
It seems that `spin` itself has `waitfor = 0` meaning you don't need to call it async here. Same for below